File: easy/70 (Climbing stairs)/solution.cpp
class Solution {
public:
    int climbStairs(int n) {
        int prev_1 = 1;
        int prev_2 = 1;
        if(n == 1) return 1;
        for(int i=2; i<=n; i++){
            int a = prev_2;
            prev_2 = prev_1 + prev_2;
            prev_1 = a;
        }
        return prev_2;
    }
};

File: easy/70 (Climbing stairs)/solution.py
class Solution:
    def climbStairs(self, n: int) -> int:
        dp = []
        dp.append(0)
        dp.append(1)
        dp.append(2)
        if n <= 2:
            return n
        
        for i in range(3, n+1, 1):
            dp.append(dp[i - 2] + dp[i - 1])
        return dp[n]